controller:
  # Fixed parameter
  DIAMETER: 0.064
  RADIUS: 0.133414
  STRRESOLUTION: 2560
  DRVRESOLUTION: 384 # [gear] from 15 to 24, 240 PPR
  # Steer Paramete
  MAX_Steer_PWM: 235 # 240くらいからオーバーシュートする
  STRKP: 140 # 150以上だと少しオーバーシュートする default=140
  STRKI: 0.01
  STRKD: 0.0
  STROFFSET: 10
  # Drive Parameter
  MAX_Drive_PWM: 80 # 現状はPMW 80くらいしか正常に動作しない
  DRVKP: 2.5 # 要パラメータ調整
  DRVKI: 0.0
  DRVKD: 0.0
  DRVOFFSET: 0
  # Others
  FRIQUENCY: 100
  LIMIT: 0.25
  ACC: 5
  STEP: 10 # FF 実装のためのプログラム
  KF: 17.38
  KV: 1.5 # 4.0がmax
  KW: 2.5

odometry: 
  FRIQUENCY: 100

MMMroute:
  FRIQUENCY: 20
  den: 500
  ahed: 15
  STDVEL: 1.2 # 1.0 m/s
  UPVEL: 1.5
  MAX_VEL: 2.0
  MIN_VEL: 0.1
  VEL_STP: 0.1
  XBORDER: 1.0
  YBORDER: -2.0
  CBORDER: 100
  UP_RANGE: 10 #[deg]
  UP_YBORDER: -1.3

SMMroute:
  FRIQUENCY: 20
  den: 500
  ahed: 15
  STDVEL: 1.1 # 1.0 m/s
  UPVEL: 1.4 # 1.6
  MAX_VEL: 2.0
  MIN_VEL: 0.1
  VEL_STP: 0.1
  UpBorderXup: 3.1
  UpBorderXdown: 1.2
  BallSetBorderX: 1.5
  BallSetBorderY: -1.5
  BallSetGoalX: -0.2
  BallSetGoalY: -3.0
  BallSetGoalR: 1.0
  StrBackGoalX: 1.0
  StrBackGoalY: -2.0
  StrBackGoalR: 0.5
  UP_RANGE: 10 #[deg]

  SMLroute:
    FRIQUENCY: 20
    den: 500
    ahed: 15
    STDVEL: 1.1 # 1.0 m/s
    UPVEL: 1.3 
    TURNVEL: 0.8 #[m/s]
    MAX_VEL: 2.0
    MIN_VEL: 0.1
    VEL_STP: 0.1
    UpBorderXup: 3.1
    UpBorderXdown: 1.2
    BallSetBorderX: 1.5
    BallSetBorderY: -1.5
    BallSetGoalX: -0.2
    BallSetGoalY: -3.0
    BallSetGoalR: 1.0
    StrBackGoalX: 1.0
    StrBackGoalY: -2.0
    StrBackGoalR: 0.5
    UP_RANGE: 10 #[deg]
    TurnRadiusNum: 10 #[個]